ansible/roles-infra/infra-ec2-template-generate/tasks/validate_instances_amis.yaml

```diff
@@ -0,0 +1,116 @@
+  - name: test 1
+    debug: msg="{{ instances }}"
+
+  - name: prep facts
+    set_fact:
+      instances_DoD: "{{ [] }}"  # DoD
+  
+  - name: fail if I can't map an image to aws_ami_region_mapping
+    loop: "{{ instances }}"
+    when:
+    - item.image is defined
+    - aws_ami_region_mapping[aws_region][item.image] is not defined
+    fail:
+      msg: >- 
+        instance.image "{{ item.image }}" is not defined in aws_ami_region_mapping.
+
+  - name: create instances_DoD from instances LoD
+    loop: "{{ instances }}"
+    loop_control:
+      extended: yes
+    set_fact: 
+      instances_DoD: >-
+        {{ 
+          instances_DoD| combine(
+            { ansible_loop.index0: instances[ansible_loop.index0]  }
+          )
+        }}
```

Comment:
Isn't this equivalent to:

```yaml
- name: create instances_DoD from instances LoD
  set_fact:
    instances_DoD: "{{ instances }}"
```

Reply:
yup

Reply:
Nope, it's going from a LIST of DICTS (LoD) to a DICT of DICTS (DoD)